products/libllbuild/BuildDB-C-API.cpp

```diff
@@ -0,0 +1,286 @@
+//
+//  BuildDB-C-API.c
+//  libllbuild
+//
+//  Copyright © 2019 Apple Inc. All rights reserved.
+//
+
+#include <llbuild/llbuild.h>
+
+#include "llbuild/Core/BuildDB.h"
+
+#include "llvm/Support/raw_ostream.h"
+
+#include <algorithm>
+#include <cassert>
+#include <memory>
+
+using namespace llbuild;
+using namespace llbuild::core;
+
+namespace {
+
+class CAPIBuildDBResultKeys {
+private:
+  std::unique_ptr<std::map<KeyID, KeyType>> keys;
```

Comment:
This probably isn't shared, so there is no need to use a std::unique_ptr, you can just store it directly

Reply:
I used unique_ptr since that provides ARC-like memory management. But I think you're right that that's not necessary as long as the object gets not shared, I'll remove it